core/trino-main/src/main/java/io/trino/sql/planner/ConnectorExpressionTranslator.java

```diff
@@ -410,6 +422,23 @@ public ConnectorToSqlExpressionTranslator(Session session, PlannerContext planne
 
             return Optional.empty();
         }
+
+        private Optional<Expression> translateIf(ConnectorExpression condition, ConnectorExpression trueValue, Optional<ConnectorExpression> falseValue)
+        {
+            Optional<Expression> conditionExpression = translate(condition);
+            Optional<Expression> trueExpression = translate(trueValue);
+
+            if (conditionExpression.isPresent() && trueExpression.isPresent()) {
+                if (falseValue.isPresent()) {
+                    Optional<Expression> falseExpression = translate(falseValue.get());
+                    return Optional.of(new IfExpression(conditionExpression.get(), trueExpression.get(), falseExpression.orElse(null)));
```

Comment:
if `falseExpression` couldn't be translated we need to return `Optional.empty()` instead of ignoring its value